src/IslandoraUtils.php

```diff
@@ -255,37 +255,43 @@ static function (string $field) {
       /* Encode the query:
        *
        * @code
-       * SELECT media__field_media_file.revision_id
-       * FROM media__field_media_file media__field_media_file
-       * WHERE media__field_media_file.field_media_file_target_id = $fid
+       * SELECT m.vid, m.mid
+       * FROM media__field_media_file f
+       * JOIN media m ON m.vid = f.revision_id
+       * JOIN media_field_data mfd ON mfd.mid = m.mid
+       * WHERE f.field_media_file_target_id = $fid
        * @endcode
        */
       $table = str_replace('.', '__', $key);
-      $subquery = $this->database->select($table, $table);
-      $subquery->addField($table, 'revision_id', 'vid');
-      $subquery->condition("{$table}.{$field}", $fid, '=');
 
-      // SELECT base_table.vid AS vid, base_table.mid AS mid
-      // FROM media base_table WHERE base_table.vid IN.
-      $queries[$key] = $this->database->select('media', 'm')
-        ->fields('m', ['mid'])
-        ->condition('m.vid', $subquery, 'IN');
+      $subquery = $this->database->select($table, 'f');
+      $subquery->join('media', 'm', 'm.vid = f.revision_id');
+      $subquery->join('media_field_data', 'mfd', 'mfd.mid = m.mid');
```

Comment:
Feel like I'm missing something: This join (with `media_field_data`) do anything? We're already joining against `media` which should only contain the default revisions. Here, we're joining against `media_field_data` which could allows access to other data fields, but we don't reference any, and it shouldn't further constrain anything (it should have a row for all of the default revisions).